airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py

```diff
@@ -83,6 +85,14 @@ class ShopifyBulkManager:
     log_job_state_msg_count: int = field(init=False, default=0)
     # one time retryable error counter
     _one_time_error_retried: bool = field(init=False, default=False)
+    # retry counter
+    _job_retry_on_error_count: int = field(init=False, default=0)
+
+    # the set of retryable errors
+    _retryable_errors: Final[set] = (
```

Comment:
The Final annotation is not very useful in making values immutable. Maybe for constants, we could use either a function that will always return the same, or we could use immutable data structures like `frozenset`.

Reply:
I don't see the value in making set > frozen set because no one will change that in the middle of the source is running. However, I'll follow your comment and apply it - it seems reasonable from the part of best practices.